be/src/runtime/local_tablets_channel.cpp

```diff
@@ -539,6 +508,49 @@ void LocalTabletsChannel::_commit_tablets(const PTabletWriterAddChunkRequest& re
     _abort_replica_tablets(request, "", node_id_to_abort_tablets);
 }
 
+void LocalTabletsChannel::_wait_secondary_replicas_commit(const PTabletWriterAddChunkRequest& request) {
+    auto start_wait_ts = butil::gettimeofday_s();
+    bool timeout = false;
+    for (auto& [tablet_id, delta_writer] : _delta_writers) {
+        // Wait util seconary replica commit/abort by primary
+        if (delta_writer->replica_state() == Secondary) {
+            do {
+                auto state = delta_writer->get_state();
+                if (state == kCommitted || state == kAborted || state == kUninitialized) {
+                    break;
+                }
+                // only sleep in bthread
+                bthread_usleep(10000); // 10ms
+
+                int64_t now = butil::gettimeofday_s();
+                auto total_rpc_time = now - _rpc_start_ts;
+                // whole rpc timeout
+                if (total_rpc_time > request.timeout_ms() / 1000) {
+                    LOG(WARNING) << "LocalTabletsChannel txn_id: " << _txn_id << " load_id: " << print_id(request.id())
+                                 << " wait tablet " << tablet_id << " secondary replica finish timeout "
+                                 << request.timeout_ms() << "ms still in state " << state;
+                    timeout = true;
+                    break;
+                }
+
+                auto wait_time = now - start_wait_ts;
+                // for secondary replica, we only wait at most secondary_replica_commit_timeout_sec second
+                // since it's the time that primary replica send commit to secondary
+                if (wait_time > std::min(request.timeout_ms() / 500, config::secondary_replica_commit_timeout_sec)) {
```

Comment:
This may change the behavior? For example, some load can success before although the `wait_time` can exceed `secondary_replica_commit_timeout_sec`, but now they will fail